composer.json

```diff
@@ -5,7 +5,8 @@
   "require": {
     "php": ">=8.2",
     "laravel/framework": ">=8.40",
-    "ramsey/uuid": "^4.1"
+    "ramsey/uuid": "^4.1",
+    "spatie/laravel-data": "^3.7"
```

Comment:
I would prefer not to depend on this package in this library because using it allows too much functionality on morphing the objects. For example, with Spatie laravel data I get `from` function, and don't think it's necessary to have it in `integration-commons` (given that it is a public package).

These extra functions for me open the door to having behavior that is not very predictable and would allow for example an `Item` class to be created in a way different than using the constructor. 

In the boundry of `integration-commons` I do not really see the value of having the Laravel Data package, the transformations we do are very simple. The benefit of them being part of the entity classes is that it is very easy to test with unit tests - what you see is what you get, no hidden behavior coming from a package.

Reply:
@M4tini to refine my argument a little bit more - one principle that is active with the current profile of `integration-commons` is how limited the domain objects are - a developer cannot do a lot with them (on purpose), they are only used to form the json responses in a correct way (strictly typed). In fact, I extracted these domain objects in `integration-commons` with the intention to make it difficult for integrators to tinker with them (thus simplifying integration development). Introducing Laravel Data does not improve anything at all from perspective of an integrator because they still will only create these read-only domain objects (like Order or Item), put them together and return them in the controllers. 

Therefore, any potential benefits are limit to the tests written in the `integration-commons` package itself. This slims down the necessity of laravel data since `toArray` is not really a bottleneck - we do not touch these functions frequently if not at all. The difference between introducing a new field in `Item` for example with a local toArray() function and without it is very minimal - one line to be precise.